### PR TITLE
Feat/profile setting

### PR DESCRIPTION
**Pull request by karljerico**
### OVERVIEW

- Created Database for profile information
- Added new route for profile information 
- Added Controller, Request and Service for profile info feature
- Fixed authorize view for the profile information

### NOTES

- Run php artisan migrate

### SNIPPETS
image
image
image
image

### CODE REVIEW

File: src/resources/views/user/settings.blade.php [hunk 1]
```diff
@@ -14,87 +14,84 @@
...
+                            </form>
                         </div>
                     </div>
                 </div>
             </div>
         </div>
-        </div>
-@endsection
+    </div>
+    @endsection
```
Comment (bpocrafael) [inline src/resources/views/user/settings.blade.php]: add new line

File: src/app/Http/Controllers/ProfileInformationController.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Contracts\View\View;
+use App\Services\ProfileInformationService;
+use App\Http\Requests\ProfileInformationStoreRequest; // Include the custom request class
```
Comment (bpocrafael) [inline src/app/Http/Controllers/ProfileInformationController.php]: is this necessary?

File: src/database/migrations/2023_10_24_080323_create_profile_information_table.php [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    /**
+     * Reverse the migrations.
+     */
+    public function down(): void
+    {
+        Schema::dropIfExists('profiles_information');
+    }
+};
```
Comment (bpocrafael) [inline src/database/migrations/2023_10_24_080323_create_profile_information_table.php]: add new line

File: src/resources/views/user/settings.blade.php (hunk 1 above)
Comment (karljerico) [reply src/resources/views/user/settings.blade.php]: New line added, Thank you sir 🙇

File: src/app/Http/Controllers/ProfileInformationController.php (hunk 2 above)
Comment (karljerico) [reply src/app/Http/Controllers/ProfileInformationController.php]: Comment is removed sir, thank you 🙇🙇

File: src/database/migrations/2023_10_24_080323_create_profile_information_table.php (hunk 3 above)
Comment (karljerico) [reply src/database/migrations/2023_10_24_080323_create_profile_information_table.php]: New line added, Thank you sir 🙇

File: src/resources/views/user/settings.blade.php [hunk 4]
```diff
@@ -14,87 +14,84 @@
                                 <div class="user-avatar mb-3">
                                     <img src="https://cdn-icons-png.flaticon.com/512/456/456283.png" alt="Profile Picture" height="250px" width="250px">
                                 </div>
-                                <h5 class="user-name">Full Name</h5>
-                                <h6 class="user-email">Email@email.com</h6>
+                                @auth
+                                    @if (auth()->user()->profileInformation)
+                                        <h5 class="user-name">
+                                            {{ auth()->user()->profileInformation->firstname ? ' ' . auth()->user()->profileInformation->firstname : '' }}
```
Comment (bpocrafael) [inline src/resources/views/user/settings.blade.php]: cant you make a accessor of this model for this one. like maybe fullName

File: src/app/Http/Controllers/ProfileInformationController.php [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+class ProfileInformationController extends Controller
+{
+    protected ProfileInformationService $profileService;
+
+    public function __construct(ProfileInformationService $profileService)
+    {
+        $this->profileService = $profileService;
+    }
+
+    public function create(): View|RedirectResponse
```
Comment (hipe01mark) [inline src/app/Http/Controllers/ProfileInformationController.php:24]: Add comment!

File: src/app/Http/Controllers/ProfileInformationController.php [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        $user = auth()->user();
+        if (!$user) {
+            return redirect()->route('login');
+        }
+        $profileInformation = $this->profileService->getProfileInformation($user);
+
+        return view('user.settings', compact('profileInformation'));
+    }
+
+    public function store(ProfileInformationStoreRequest $request): RedirectResponse
```
Comment (hipe01mark) [inline src/app/Http/Controllers/ProfileInformationController.php:36]: Add comment!

File: src/app/Services/ProfileInformationService.php [hunk 7]
```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace App\Services;
+
+use App\Models\User;
+use App\Models\ProfileInformation;
+
+class ProfileInformationService
+{
+    public function getProfileInformation(User $user): ProfileInformation
```
Comment (hipe01mark) [inline src/app/Services/ProfileInformationService.php]: Add comment!

**Review by hipe01mark — DISMISSED** [review]
LGTM 🙇

File: src/routes/web.php [hunk 8]
```diff
@@ -29,6 +29,7 @@
     Route::get('/comments/{comment}/edit', [CommentController::class, 'edit'])->name('comments.edit');
     Route::put('/comments/{comment}', [CommentController::class, 'update'])->name('comments.update');
     Route::delete('/comments/{comment}', [CommentController::class, 'destroy'])->name('comments.destroy');
+    Route::resource('profileinfo', ProfileInformationController::class);
```
Comment (bpocrafael) [inline src/routes/web.php]: two words? it should be
```suggestion
    Route::resource('profile-info', ProfileInformationController::class);
```

Comment (karljerico) [reply src/routes/web.php]: ow, Thank you sir didn't know that. Will change it po 🙇🙇🙇

**Review by bpocrafael — DISMISSED** [review]

**Review by hipe01mark — APPROVED** [review]
LGTM 🙇

**Review by bpocrafael — APPROVED** [review]

**Pull request merged**